### PR TITLE
feat(recovery): Store and load wallet credentials in metadata

### PR DESCRIPTION
**Pull request by matsjj**
## Description (optional)
Going forward, with the first login, the wallet will store the credentials for login, as in, guid, password and sharedKey in metadata. As such, using the mnemonic it will be possible to login to an existing wallet, even when 2FA has been lost. Doing it this way will keep all information related to nabu and the custodial part of the wallet, as the user is literally logging into the very same wallet. Previously it would create a new wallet instead with the old mnemonic. 

## Testing Steps (optional)
Happy to write a few tests if someone gives me a pointer on what should be tested ideally. 

For manual testing, use an old wallet, go through mnemonic recovery and discover that the GUID has changed. Upon logging in with this build, going through the same steps you should discover that the GUID has remained the same

### CODE REVIEW

**Review by chtaili77 — APPROVED** [review]

**Comment by Supattr0712** [comment]
ากัสื

**Comment by Supattr0712** [comment]
ย้ดสา

**Comment by Supattr0712** [comment]
ดร่ิ

**Comment by Supattr0712** [comment]
Bfgkkcduhc

**Comment by Supattr0712** [comment]
Hdkgdjjv

**Comment by Supattr0712** [comment]
Uhchkvc

**Comment by Supattr0712** [comment]
Bxhjc

**Comment by Supattr0712** [comment]
Jdtjgc

**Comment by Supattr0712** [comment]
Nxhkcfh

**Comment by Supattr0712** [comment]
Bxhlvcv

**Comment by Supattr0712** [comment]
Gdhkfd

**Comment by Supattr0712** [comment]
Gdhkfd

**Comment by Supattr0712** [comment]
Nxhknvff

**Comment by Supattr0712** [comment]
Hcgkhx

**Comment by Supattr0712** [comment]
[]()

**Comment by Supattr0712** [comment]
[]()

**Comment by Supattr0712** [comment]
[]()

**Comment by Supattr0712** [comment]
[]()

**Comment by Supattr0712** [comment]
Hfghv

**Comment by Supattr0712** [comment]
Ncfhh

**Comment by Supattr0712** [comment]
Hdfhjx

**Comment by Supattr0712** [comment]
Hfgjkgd

**Comment by Supattr0712** [comment]
John fu jbv

**Comment by Supattr0712** [comment]
Hchjf

**Comment by Supattr0712** [comment]
Bfgjh

**Comment by Supattr0712** [comment]
Bcdthg

**Comment by Supattr0712** [comment]
Uffh

**Comment by Supattr0712** [comment]
High kg

**Comment by Supattr0712** [comment]
Hdgkff

**Comment by Supattr0712** [comment]
Jbufdu

**Comment by Supattr0712** [comment]
Jvjfjd

**Comment by Supattr0712** [comment]
Nbfyg

**Comment by Supattr0712** [comment]
Kfdyhf

**Comment by Supattr0712** [comment]
Jfdfg

**Comment by Supattr0712** [comment]
Jfdgj

**Comment by Supattr0712** [comment]
Jcdgj

**Comment by Supattr0712** [comment]
Jcdhj

**Comment by Supattr0712** [comment]
Hdfh

**Comment by Supattr0712** [comment]
Hfdg

**Comment by Supattr0712** [comment]
Hffkh

**Comment by Supattr0712** [comment]
Bdfjg

**Comment by Supattr0712** [comment]
'Dfh

File: packages/blockchain-wallet-v4-frontend/src/assets/locales/es.json [hunk 1]
```diff
@@ -4956,4 +4956,4 @@
   "wallettour.airdrop.tooltip.stx": "Nuestro último airdrop con Blockstack ya está aquí. Asegúrese de conseguir sus Stacks (STX) gratis hoy.",
```
Comment (yichok) [inline packages/blockchain-wallet-v4-frontend/src/assets/locales/es.json:4956]: packages/blockchain-wallet-v4-frontend/src/assets/locales/es.json

**Review by yichok — CHANGES_REQUESTED** [review]

**Comment by yichok** [comment]
> ## Description (optional)
> Going forward, with the first login, the wallet will store the credentials for login, as in, guid, password and sharedKey in metadata. As such, using the mnemonic it will be possible to login to an existing wallet, even when 2FA has been lost. Doing it this way will keep all information related to nabu and the custodial part of the wallet, as the user is literally logging into the very same wallet. Previously it would create a new wallet instead with the old mnemonic.
> 
> ## Testing Steps (optional)
> Happy to write a few tests if someone gives me a pointer on what should be tested ideally.
> 
> For manual testing, use an old wallet, go through mnemonic recovery and discover that the GUID has changed. Upon logging in with this build, going through the same steps you should discover that the GUID has remained the same

**Review by yichok — APPROVED** [review]

**Review by yichok — APPROVED** [review]

File: packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/actionTypes.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export const FETCH_METADATA_WALLET_CREDENTIALS_SUCCESS = '@CORE.FETCH_METADATA_WALLET_CREDENTIALS_SUCCESS'
+export const FETCH_METADATA_WALLET_CREDENTIALS_FAILURE = '@CORE.FETCH_METADATA_WALLET_CREDENTIALS_FAILURE'
+
+// UPDATE_METADATA_WALLET_CREDENTIALS
+export const UPDATE_METADATA_WALLET_CREDENTIALS = '@CORE.KVSTORE.UPDATE_METADATA_WALLET_CREDENTIALS'
+
+// CREATE_METADATA_WALLET_CREDENTIALS
+export const CREATE_METADATA_WALLET_CREDENTIALS = '@CORE.CREATE_METADATA_WALLET_CREDENTIALS'
+
+export const SET_TRANSACTION_NOTE_WALLET_CREDENTIALS = '@CORE.SET_TRANSACTION_NOTE_WALLET_CREDENTIALS'
```
Comment (mpfluger) [inline packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/actionTypes.js:13]: copy-pasta error?

File: packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/sagas.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import * as A from './actions'
+import { callTask } from '../../../utils/functional'
+import { derivationMap, WALLET_CREDENTIALS } from '../config'
+// import { getKeyPair } from '../../../utils/walletCredentials'
+import * as selectors from '../../selectors'
+import { getMetadataXpriv } from '../root/selectors'
+import { KVStoreEntry } from '../../../types'
+
+export default ({ api, networks } = {}) => {
+  const createWalletCredentials = function * (kv) {
```
Comment (mpfluger) [inline packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/sagas.js:13]: Maybe rename, `updateWalletCredentialsIfChangedOrMissing`

File: packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/sagas.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    try {
+      const typeId = derivationMap[WALLET_CREDENTIALS]
+      const mxpriv = yield select(getMetadataXpriv)
+      const kv = KVStoreEntry.fromMetadataXpriv(mxpriv, typeId, networks.btc)
+      yield put(A.fetchMetadataWalletCredentialsLoading())
+      const newkv = yield callTask(api.fetchKVStore(kv))
+      yield call(createWalletCredentials, newkv)
+    } catch (e) {
+      // eslint-disable-next-line
+      console.error(e)
```
Comment (mpfluger) [inline packages/blockchain-wallet-v4/src/redux/kvStore/walletCredentials/sagas.js:48]: Maybe not log the whole error

**Review by schnogz — APPROVED** [review]

**Pull request merged**